forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/repository/migration/v3_1_6/MongoDbPaymentsChangeLog.java

```diff
@@ -24,83 +24,153 @@
+    }
 
-        CloseableIterator<FRInternationalScheduledPaymentSubmission2> frInternationalScheduledPaymentSubmissions = getLegacyDocuments(mongoTemplate, FRInternationalScheduledPaymentSubmission2.class);
-        frInternationalScheduledPaymentSubmissions.forEachRemaining(d -> migrate(mongoTemplate, d, toFRInternationalScheduledPaymentSubmission(d)));
+    /*
+     * List of objects to update
+     * @return list of objects to update
+     */
+    private List<ObjectToUpdate> getObjectsToUpdate(){
+        List<ObjectToUpdate> objectsToUpdate = new ArrayList<>();
+        // Domestic payments
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticConsent.class, "domesticConsent"));
+
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticPaymentSubmission.class, "domesticPayment"));
+        // Domestic scheduled payments
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticScheduledConsent.class, "domesticScheduledConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticScheduledPaymentSubmission.class, "domesticScheduledPayment"));
+        // Domestic standing orders
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticStandingOrderConsent.class, "domesticStandingOrderConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticStandingOrderPaymentSubmission.class, "domesticStandingOrder"));
+        // International payments
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalConsent.class,"internationalConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalPaymentSubmission.class, "internationalPayment"));
+        // International scheduled payments
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalScheduledConsent.class, "internationalScheduledConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalScheduledPaymentSubmission.class, "internationalScheduledPayment"));
+        // International standing Orders
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalStandingOrderConsent.class, "internationalStandingOrderConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalStandingOrderPaymentSubmission.class, "internationalStandingOrder"));
+
+        objectsToUpdate.add(new ObjectToUpdate(FRPaymentSetup.class, "paymentSetupRequest"));
+
+
+        return objectsToUpdate;
+    }
 
-        CloseableIterator<FRInternationalStandingOrderConsent3> frInternationalStandingOrderConsents = getLegacyDocuments(mongoTemplate, FRInternationalStandingOrderConsent3.class);
-        frInternationalStandingOrderConsents.forEachRemaining(d -> migrate(mongoTemplate, d, toFRInternationalStandingOrderConsent(d)));
+    class ObjectToUpdate{
+        Class legacyClass; // class from legacy.payments
+        String writeRiskParentField; // name of the field to get FRWrite object that contains the Risk and root field of document to path risk elements, stored in mongodb
```

Comment:
You can ignore this comment - it's just for interest/discussion...

I've worked with a lot of developers who would hate this lack of type safety! One option is to make it a `Field` but you'd still need to construct it from a String. For example:

```
private Class<? extends LegacyCountrySubDivision> legacyClass;
private Field frWriteField;

        public ObjectToUpdate(Class<? extends LegacyCountrySubDivision> legacyClass, String frWriteField) {
            this.legacyClass = legacyClass;
            try {
                this.frWriteField = legacyClass.getDeclaredField(frWriteField);
            } catch (NoSuchFieldException e) {
                throw new IllegalArgumentException(String.format("Cannot find FRWrite field with name [%s] on class [%s]", frWriteField, legacyClass.getSimpleName()));
            }
        }
```
This has the advantage of failing early, but it does't give much type safety.

If it was my choice then I'd probably remove this class altogether and just have `getObjectsToUpdate()` return `List<Class<? extends LegacyCountrySubDivision>>`. However, you'd need to wildcard the FRWrite field in your JSON path, which you may not like 😄 🤔